src/lib/remark-plugins/remark-plugin-adjust-link-urls/helpers.ts

```diff
@@ -94,10 +123,16 @@ const preAdjustUrl = ({ currentPath, url }): string => {
 		return url
 	}
 
-	// Do nothing if url is not an internal path
-	const isInternalPath = getIsInternalPath(url)
-	if (!isInternalPath) {
-		return url
+	// Handle developer.hashicorp.com links if needed
+	try {
+		const { hostname, pathname, search = '', hash = '' } = new URL(url)
+		if (hostname === 'developer.hashicorp.com') {
+			return `${pathname}${search}${hash}`
```

Comment:
Do you want to return the path here? Wouldn't you want to keep rewriting it?

Reply:
This function only pre-adjusts URLs before they're actually rewritten. The `remarkPluginAdjustLinkUrls` calls `processDocsLinkNode` (defined at the top of this file), which calls `preAdjustUrl`, then passes the result of `preAdjustUrl` to `urlAdjustFn` (which does the rewriting).

Reply:
No, because there's nothing to adjust for top-level paths. This can be seen in the next `if` block after this `tr/catch`:

```ts
// Do nothing if url is a top-level path
if (url.startsWith('/')) {
  return url
}
```

I'll add some more comments for clarity!

Reply:
Added a `handleFullUrl` helper in https://github.com/hashicorp/dev-portal/pull/1544/commits/b0ba9d6a739e48aed9b7ff17b137c4afda1a616e!